src/planwise/client/scenarios/views.cljs

```diff
@@ -233,16 +233,17 @@
           [:small "Computing best locations ..."]])])))
 
 (defn display-current-scenario
-  [current-project current-scenario]
+  [current-project {:keys [id] :as current-scenario}]
   (let [read-only? (subscribe [:scenarios/read-only?])
         state      (subscribe [:scenarios/view-state])
         error      (subscribe [:scenarios/error])
         created-providers (subscribe [:scenarios/created-providers])
         source-demand (get-in current-project [:engine-config :source-demand])
         unit-name  (get-in current-project [:config :demographics :unit-name])
+        csv-name (str (:id current-project) "-" id "-" (:name current-scenario) ".csv")
         export-providers-button [:a {:class "mdc-fab disable-a"
                                      :id "main-action"
-                                     :href (str "/api/scenarios/" (:id current-scenario) "/csv")}
+                                     :href (str "/api/scenarios/" id "/" csv-name)}
```

Comment:
Constructing a URL from a user input is generally a bad idea, as it may lead to malformed URLs, malicious attacks and data leakage. Output file name can be set via a HTTP header in the response instead.